unit-tests/utilities/string/test-string-to-value.cpp

```diff
@@ -0,0 +1,154 @@
+    us_tests.push_back(string_checker<unsigned short>{ "123", 123, true });
+    us_tests.push_back(string_checker<unsigned short>{ "-123", 0, false });
+    us_tests.push_back(string_checker<unsigned short>{ "123456789123456789123456789", 0, false });
+    us_tests.push_back(string_checker<unsigned short>{ "-123456789123456789123456789", 0, false });
+    us_tests.push_back(string_checker<unsigned short>{ "abc", 0, false });
+
+    check_tests<unsigned short>(us_tests);
+
+    // long
+    std::vector<string_checker<long>> l_tests;
+    l_tests.push_back(string_checker<long>{ "123", 123, true });
+    l_tests.push_back(string_checker<long>{ "-123", -123, true });
+    l_tests.push_back(string_checker<long>{ "-123456789", -123456789, true });
+    l_tests.push_back(string_checker<long>{ "123456789123456789123456789", 0, false });
+    l_tests.push_back(string_checker<long>{ "-123456789123456789123456789", 0, false });
+    l_tests.push_back(string_checker<long>{ "abc", 0, false });
+
+    check_tests<long>(l_tests);
+
+
+    // long long
+    std::vector<string_checker<long long>> ll_tests;
+    ll_tests.push_back(string_checker<long long>{ "123", 123, true });
+    ll_tests.push_back(string_checker<long long>{ "-123", -123, true });
+    ll_tests.push_back(string_checker<long long>{ "12345", 12345, true });
+    ll_tests.push_back(string_checker<long long>{ "-12345", -12345, true });
+    ll_tests.push_back(string_checker<long long>{ "123456789123456789123456789", 0, false });
+    ll_tests.push_back(string_checker<long long>{ "-123456789123456789123456789", 0, false });
+    ll_tests.push_back(string_checker<long long>{ "abc", 0, false });
+
+    check_tests<long long>(ll_tests);
+
+    // ungined long long
+    std::vector<string_checker<unsigned long long>> ull_tests;
+    ull_tests.push_back(string_checker<unsigned long long>{ "123", 123, true });
+    ull_tests.push_back(string_checker<unsigned long long>{ "123456789", 123456789, true });
+    ull_tests.push_back(string_checker<unsigned long long>{ "-123456789", 0, false });
+    ull_tests.push_back(string_checker<unsigned long long>{ "123456789123456789123456789", 0, false });
+    ull_tests.push_back(string_checker<unsigned long long>{ "abc", 0, false });
+
+    check_tests<unsigned long long>(ull_tests);
+    
+    // float
+    std::vector<string_checker<float>> f_tests;
+    f_tests.push_back(string_checker<float>{ "1.23456789", 1.23456789, true });
+    f_tests.push_back(string_checker<float>{ "2.12121212", 2.12121212, true });
+    f_tests.push_back(string_checker<float>{ "-1.23456789", -1.23456789, true });
+    f_tests.push_back(string_checker<float>{ "-2.12121212", -2.12121212, true });
+    f_tests.push_back(string_checker<float>{ "INF", 0, false });
+    f_tests.push_back(string_checker<float>{ "-INF", 0, false });
+    f_tests.push_back(string_checker<float>{ "NaN", 0, false });
+    f_tests.push_back(string_checker<float>{ "abc", 0, false });
```

Comment:
For float/double:
1. add test for subnormal values:
`"isnormal(DBL_MIN/2.0) = " << std::isnormal(DBL_MIN/2.0)`
2. Check overflow test ( x > max_float, x< -max_float)